theseus/optimizer/nonlinear/nonlinear_optimizer.py

```diff
@@ -125,6 +129,22 @@ def _init_info(
             err_history[:, 0] = last_err.clone().cpu()
         else:
             err_history = None
+
+        if track_state_history:
+            state_history = {}
+            for var in self.objective.optim_vars.values():
+                state_history[var.name] = (
+                    torch.ones(
+                        self.objective.batch_size,
+                        var.dof(),
+                        self.params.max_iterations + 1,
+                    )
+                    * math.inf
```

Comment:
Is torch.empty better than inf?

Reply:
`torch.empty` will most likely initialize the tensor with pretty much random values, so it'll be very hard to figure out automatically what the EOS character is. I agree that `inf` is probably not great, but I'm not sure if this is much of an issue.

Reply:
Okay in that case we can just clarify in documentation since any non inf values will also be ambiguous.